cardano-api/src/Cardano/Api/ProtocolParameters.hs

```diff
@@ -212,14 +226,380 @@ data ProtocolParameters =
        --
        -- This is the \"tau\" incentives parameter from the design document.
        --
-       protocolParamTreasuryCut :: Rational
+       protocolParamTreasuryCut :: Rational,
+
+       -- Introduced in Alonzo
+
+       -- | Cost in ada per byte of UTxO storage (instead of
+       --protocolParamMinUTxOValue in the Alonzo era onwards).
+       protocolParamUTxOCostPerByte :: Maybe Lovelace,
+
+       -- | Cost models for non-native script languages.
+       protocolParamCostModels :: Maybe CostModel,
```

Comment:
This is the collection of all cost models.

So I think we should use `Map AnyScriptLanguage CostModel` here directly. It can be empty so we would not need an extra `Maybe`.